### DISCUSSION TITLE
working Choose Action examples?

### DISCUSSION THREAD

**Discussion by tomByrer** [statelyai/xstate#1810] [Q&A]
Does anyone have working [Choose Action](https://xstate.js.org/docs/guides/actions.html#choose-action) examples please?

I keep getting `TypeError: Object(...) is not a function` error in my local project, & I can't even get a simple vis example due to `ReferenceError: choose is not defined`

**Comment by danrocha** [comment]
Can you show us your code? It sounds like your actions are either not defined properly or are in the wrong location...

**Comment by tomByrer** [comment]
Well, it seems `choose` is not exported to the vis:
```js
  // Available variables:
  // - Machine
  // - interpret
  // - assign
  // - send
  // - sendParent
  // - spawn
  // - raise
  // - actions
  // - XState (all XState exports)
```
But there are quite a few [tests](https://github.com/davidkpiano/xstate/blob/161171007334abecb787183e99acb980aaee079a/packages/core/test/actions.test.ts#L1040) I can look at.

**Reply by davidkpiano** [reply]
The visualizer is currently running an older version of XState. The inspector always uses the latest version of XState.

**Reply by tomByrer** [reply]
I was hoping to improve the docs with an inline vis example.  Can't really do that with CodePen?  + CodePen tends to take long to boot & eat extra resources; at that point I'd rather just use my local editor.

**Reply by davidkpiano** [reply]
Good point; eventually, the visualizer will be updated to use a lot of parts from the inspector so that you can share inline viz.
